src/components/app/ProfileViewer.js

```diff
@@ -53,6 +53,31 @@ type DispatchProps = {|
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class ProfileViewer extends PureComponent<Props> {
+  manageBeforeUnloadListener() {
+    const { isUploading } = this.props;
+    if (isUploading) {
+      window.addEventListener('beforeunload', this.handleUnload);
+    } else {
+      window.removeEventListener('beforeunload', () => this.handleUnload);
+    }
+  }
+
+  handleUnload = (event: BeforeUnloadEvent) => {
+    event.preventDefault();
```

Comment:
According to MDN ([see the article](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event)) some browsers (including Chrome) don't work with `preventDefault`. As a result it's advised to use one of the legacy methods too...
My suggestion would be to just add this at the end:
```js
// Return a string for legacy browsers (including Chrome). It won't likely be displayed to the user.
return 'Are you sure you want to close while uploading?';
```